### PR TITLE
HID: Update the HID service to match more closely to switchbrew part 1

### PR DESCRIPTION
**Pull request by german77**
Updates the HID service to match more closely to switchbrew. Functionality wise nothing should be changed.

- Adds missing controller types and properties.
- Renames properties names to match switchbrew

### CODE REVIEW

File: src/core/hle/service/hid/controllers/keyboard.h [hunk 1]
```diff
@@ -31,12 +32,28 @@ class Controller_Keyboard final : public ControllerBase {
     void OnLoadInputDevices() override;
 
 private:
+    struct Modifiers {
+        union {
+            s32_le raw{};
```
Comment (lioncash) [inline src/core/hle/service/hid/controllers/keyboard.h:37]:
```suggestion
            u32_le raw{};
```

File: src/core/hle/service/hid/controllers/mouse.h [hunk 2]
```diff
@@ -30,6 +31,27 @@ class Controller_Mouse final : public ControllerBase {
     void OnLoadInputDevices() override;
 
 private:
+    struct Buttons {
+        union {
+            s32_le raw{};
```
Comment (lioncash) [inline src/core/hle/service/hid/controllers/mouse.h:36]:
```suggestion
            u32_le raw{};
```

File: src/core/hle/service/hid/controllers/mouse.h [hunk 3]
```diff
@@ -30,6 +31,27 @@ class Controller_Mouse final : public ControllerBase {
...
+            BitField<2, 1, u32> middle;
+            BitField<3, 1, u32> forward;
+            BitField<4, 1, u32> back;
+        };
+    };
+    static_assert(sizeof(Buttons) == 0x4, "Buttons is an invalid size");
+
+    struct Attributes {
+        union {
+            s32_le raw{};
```
Comment (lioncash) [inline src/core/hle/service/hid/controllers/mouse.h:48]:
```suggestion
            u32_le raw{};
```

File: src/core/hle/service/hid/controllers/xpad.h [hunk 4]
```diff
@@ -28,6 +29,67 @@ class Controller_XPad final : public ControllerBase {
     void OnLoadInputDevices() override;
 
 private:
+    struct Attributes {
+        union {
+            s32_le raw{};
```
Comment (lioncash) [inline src/core/hle/service/hid/controllers/xpad.h:34]:
```suggestion
            u32_le raw{};
```

File: src/core/hle/service/hid/controllers/xpad.h [hunk 5]
```diff
@@ -28,6 +29,67 @@ class Controller_XPad final : public ControllerBase {
...
+            BitField<0, 1, u32> IsConnected;
```
Comment (lioncash) [inline src/core/hle/service/hid/controllers/xpad.h:35]:
```suggestion
            BitField<0, 1, u32> is_connected;
```

etc.

All struct members should be snake_case.

**Review by lioncash — COMMENTED** [review]
The first commit should probably be more explanatory about what was changed.

File: src/core/hle/service/hid/controllers/keyboard.h [hunk 6]
```diff
@@ -31,12 +32,28 @@ class Controller_Keyboard final : public ControllerBase {
...
+            BitField<0, 1, u32> control;
+            BitField<1, 1, u32> shift;
+            BitField<2, 1, u32> left_alt;
+            BitField<3, 1, u32> right_alt;
+            BitField<4, 1, u32> gui;
+            BitField<8, 1, u32> caps_lock;
+            BitField<9, 1, u32> scroll_lock;
+            BitField<10, 1, u32> num_lock;
+            BitField<11, 1, u32> katana;
```
Comment (Morph1984) [inline src/core/hle/service/hid/controllers/keyboard.h:46]:
```suggestion
            BitField<11, 1, u32> katakana;
```

File: src/core/hle/service/hid/controllers/npad.h [hunk 7]
```diff
@@ -337,32 +375,54 @@ class Controller_NPad final : public ControllerBase {
...
         union {
             s64_le raw{};
+            BitField<0, 1, s64> is_charging_joy_dual;
+            BitField<1, 1, s64> is_charging_joy_left;
+            BitField<2, 1, s64> is_charging_joy_right;
+            BitField<3, 1, s64> is_powered_joy_dual;
+            BitField<4, 1, s64> is_powered_joy_left;
+            BitField<5, 1, s64> is_powered_joy_right;
+            BitField<9, 1, s64> is_system_unsuported_button;
+            BitField<10, 1, s64> is_system_ext_unsuported_button;
```
Comment (Morph1984) [inline src/core/hle/service/hid/controllers/npad.h:388]:
```suggestion
            BitField<9, 1, s64> is_system_unsupported_button;
            BitField<10, 1, s64> is_system_ext_unsupported_button;
```

File: src/core/hle/service/hid/controllers/npad.h [hunk 8]
```diff
@@ -440,7 +497,7 @@ class Controller_NPad final : public ControllerBase {
     NpadHoldType hold_type{NpadHoldType::Vertical};
     NpadHandheldActivationMode handheld_activation_mode{NpadHandheldActivationMode::Dual};
     // NpadCommunicationMode is unknown, default value is 1
-    NpadCommunicationMode communication_mode{NpadCommunicationMode::Unknown1};
+    NpadCommunicationMode communication_mode{NpadCommunicationMode::Default};
```
Comment (Morph1984) [inline src/core/hle/service/hid/controllers/npad.h:500]: comment can be removed

**Comment by german77** [comment]
Needs more RE. Found some games that do not behave correctly

**Review by bunnei — APPROVED** [review]

**Pull request merged**